code-challenges.test.js

```diff
@@ -13,40 +15,111 @@
 // --------------------1) Create a function that takes in a string and returns a string with a coded message. The coded message converts "a" to 4, "e" to 3, "i" to 1, and "o" to 0.
 
 // a) Create a test with expect statements using the variables provided.
-
-const secretCodeWord1 = "Lackadaisical"
-// Expected output: "L4ck4d41s1c4l"
-const secretCodeWord2 = "Gobbledygook"
-// Expected output: "G0bbl3dyg00k"
-const secretCodeWord3 = "Eccentric"
-// Expected output: "3cc3ntr1c"
+describe('coder', ()=>{
+    it('function that takes in a string and returns a string with a coded message. The coded message converts "a" to 4, "e" to 3, "i" to 1, and "o" to 0.', ()=>{
+        const secretCodeWord1 = "Lackadaisical"
+        // Expected output: "L4ck4d41s1c4l"
+        const secretCodeWord2 = "Gobbledygook"
+        // Expected output: "G0bbl3dyg00k"
+        const secretCodeWord3 = "Eccentric"
+        // Expected output: "3cc3ntr1c"
+        expect(coder(secretCodeWord1)).toEqual("L4ck4d41s1c4l")
+        expect(coder(secretCodeWord2)).toEqual("G0bbl3dyg00k")
+        expect(coder(secretCodeWord3)).toEqual("3cc3ntr1c")
+        
+    })
+})
+
+// ReferenceError: coder is not defined
+
+// Test Suites: 1 passed, 1 total
+// Tests:       1 passed, 1 total
 
 // b) Create the function that makes the test pass.
 
+//create a function that takes a string as an argument
+const coder = (string) => {
+    //turn string into an array map over each ltr and replace it with the correct letter if it meets the right criteria then join it back together as a string to return
```

Comment:
There is some room to expand on your pseudocode here.  For instance, there is no mention of using regex or .test.